src/ruby/puppetserver-lib/puppet/server/ast_compiler.rb

```diff
@@ -34,52 +34,72 @@ def self.compile_ast(code, compile_options, boltlib_path)
           end
 
           load_bolt()
+          if compile_options.has_key?('versioned_project')
+            compile_for_project_plan(code, compile_options, boltlib_path)
+          else
+            compile_for_environment_plan(code, compile_options, boltlib_path)
+          end
+        else
+          compile_for_environment(code, compile_options, boltlib_path)
+        end
+      end
+      private_class_method :compile_ast
 
-          Puppet[:node_name_value] = compile_options['certname']
+      def self.compile_for_project_plan(code, compile_options, boltlib_path)
+        Puppet[:node_name_value] = compile_options['certname']
 
-          # Prior to PE-29443 variables were in a hash. Serialization between ruby/clojure/json did
-          # not preserver hash order necessary for deserialization. The data strucutre is now stored
-          # in a list for moving data and the list is used to construct an ordered ruby hash.
-          plan_variables = if compile_options['variables']['values'].is_a?(Array)
-                        compile_options['variables']['values'].each_with_object({}) do |param_hash, acc|
-                          acc[param_hash.keys.first] = param_hash.values.first
-                        end
-                      else
-                        compile_options['variables']['values']
-                      end
+        bolt_project = Struct.new(:name, :path, :load_as_module?).new(compile_options['project_name'],
+                                                                      compile_options['project_root'],
+                                                                      true)
+        # TODO: Given we hide this from plan authors this current iteration has only
+        # the "required" data for now. Once we can get https://github.com/puppetlabs/bolt/pull/1770
+        # merged and promoted we can just use an empty hash.
+        fake_config = {
+          'transport' => 'redacted',
+          'transports' => {
+            'redacted' => 'redacted'
+          }
+        }
+        bolt_inv = Bolt::ApplyInventory.new({fake_config})
```

Comment:
```suggestion
        bolt_inv = Bolt::ApplyInventory.new(fake_config)
```
Same in the other construction of a bolt_inv.